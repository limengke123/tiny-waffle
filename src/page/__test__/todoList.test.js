import React from 'react'
import { shallow, render } from 'enzyme'
import TodoList from '../todoList'

describe('todoList', () => {
    beforeEach(() => {
        TodoList.index = 0
    })

    it('should render correctly', () => {
        const wrapper = render(<TodoList />)
        expect(wrapper).toMatchSnapshot()
    })

    it('should add item after trigger addTodoListItem', () => {
        const wrapper = shallow(<TodoList />)
        const instance = wrapper.instance()
        instance.state = {
            todos: [],
            inputValue: '测试文字'
        }
        expect(TodoList.index).toBe(0)
        instance.addTodoListItem()
        expect(instance.state).toMatchObject({
            inputValue: '',
            todos: [
                {
                    isComplete: false,
                    id: 0,
                    info: {
                        text: '测试文字'
                    }
                }
            ]
        })
        expect(TodoList.index).toBe(1)
        instance.state.inputValue = '测试加的第二项'
        instance.addTodoListItem()
        expect(instance.state).toMatchObject({
            inputValue: '',
            todos: [
                {
                    isComplete: false,
                    id: 0,
                    info: {
                        text: '测试文字'
                    }
                },
                {
                    isComplete: false,
                    id: 1,
                    info: {
                        text: '测试加的第二项'
                    }
                }
            ]
        })
        expect(TodoList.index).toBe(2)
    })

    it('should delete item after trigger deleteTodoListItem', () => {
        const wrapper = shallow(<TodoList />)
        const instance = wrapper.instance()
        instance.state.todos = [{ id: 0 }, { id: 1 }, { id: 2 }, { id: 3 }]
        instance.deleteTodoListItem({ id: 2 })
        expect(instance.state.todos).toHaveLength(3)
        instance.deleteTodoListItem({ id: 1 })
        expect(instance.state.todos).toHaveLength(2)
    })

    it('should toggle complete when trigger handleComplete', () => {
        const wrapper = shallow(<TodoList />)
        const instance = wrapper.instance()
        instance.state.todos = [{ id: 0, isComplete: false }]

        instance.handleComplete({ id: 0 })
        expect(instance.state.todos[0].isComplete).toBeTruthy()

        instance.handleComplete({ id: 0 })
        expect(instance.state.todos[0].isComplete).toBeFalsy()
    })
})
