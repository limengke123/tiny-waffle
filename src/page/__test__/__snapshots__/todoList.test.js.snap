// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todoList should render correctly 1`] = `
<div>
  <div>
    <input
      class="todo-input"
      placeholder="请输入待办项"
      type="text"
      value=""
    />
  </div>
  <span>
    暂无数据
  </span>
</div>
`;
